### DISCUSSION TITLE
Problems with fetching data

### DISCUSSION THREAD

**Discussion by wolfedale** [nuxt/nuxt#16894] [General]
Hi everyone!

I'm completely new with Nuxt3 and also with Frontend. I started to build small POC project to see how Nuxt 3 works and I'm having issues to figure fetching data from my REST API. Somehow It's not clear to me what's the problem and how to solve it.

This is what I'm trying to do:

1. I have `components/Post/ListById.vue` which contain:
```
<template>
    <div>
            <PostItem :post="data" />
    </div>
</template>

<script setup>
    const props = defineProps({
        id: {
            type: String,
            required: true
        }
    })

    const { data } = await useAsyncData(() => $fetch('http://localhost:5000/api/v1/posts/'+props.id, {
        headers: useRequestHeaders(['cookie'])
    }))
</script>
```

2. My "PostItem" is in `components/Post/Item/index.vue`:
```
<template>
    <div>
        <PostItemHeader :post="post"/>
                {{ props.post.message }}
                <Nuxt-Link :to="/posts/ + props.post.id">read</Nuxt-Link>
    </div>
</template>

<script setup>
    const props = defineProps({
        post: {
            type: Object,
            required: true
        }
    })
</script>
```

3. My `pages/posts/[id].vue`
```
<template>
    <div>
        <MainSection title="Home" :loading="loading">

            <Head>
                <Title>Post by ID</Title>
            </Head>

            <PostListById :id=route.params.id />

        </MainSection>
    </div>
</template>

<script setup>
const loading = ref(false)
const route = useRoute();
</script>
```

When I'm going directly to `localhost:3000/posts/60` I see the post. When I'm going to post by clicking on `read` -> ` <Nuxt-Link :to="/posts/ + props.post.id">read</Nuxt-Link>` I see some errors and when I refresh my browser I'm able to see my post again.

This is the error:
```
[Warning] [Vue warn]: Invalid prop: type check failed for prop "id". Expected String with value "undefined", got Undefined  (24) (chunk-7ZDQWV6W.js, line 1086)
"
"
" at <ListById"
"id=undefined"
">"
"
"
" at <MainSection"
"title=\"Home\""
"loading=false"
">"
"
"
" at <[id]"
"onVnodeUnmounted=fn<onVnodeUnmounted>"
(...)
[Error] Unhandled Promise Rejection: TypeError: null is not an object (evaluating '$setup.props.post.message')
```

Any idea what I'm missing? :-(

**Comment by wolfedale** [comment]
I noticed that everything is working fine when I'm using simply `<a href="` instead of `<Nuxt-Link>`. So something must be wrong with Nuxt-Link and passing parameters...

**Comment by danielroe** [comment]
It should be

```vue
<Nuxt-Link :to="'/posts/' + props.post.id">
```

**Comment by wolfedale** [comment]
Doesn't work with `' ' ` too :-(

**Reply by TheAlexLichter** [reply]
`<NuxtLink>` or `<nuxt-link>`. Also, feel free to upload a reproduction so it is easier to tackle the problem (ideally, use https://nuxt.new for it ☺️)
